openedx/core/djangoapps/discussions/serializers.py

```diff
@@ -187,13 +199,11 @@ def to_internal_value(self, data: dict) -> dict:
         """
         Transform the *incoming* primitive data into a native value.
         """
-        payload = {
-            'context_key': data.get('course_key', ''),
-            'enabled': data.get('enabled', False),
+        payload = super().to_internal_value(data)
+        payload.update({
             'lti_configuration': data.get('lti_configuration', {}),
             'plugin_configuration': data.get('plugin_configuration', {}),
-            'provider_type': data.get('provider_type', DEFAULT_PROVIDER_TYPE),
```

Comment:
Why is `provider_type` removed?

Reply:
The key is in the payload. We have `payload = super().to_internal_value(data)` which is using DRF's automatic mechanism to handle `provider_type`. Since `lti_configuration` and `plugin_configuration` are nested representations, DRF can't handle that directly, so we are processing them here while leaving the rest to DRF.

Reply:
got it.